src/include/transaction/deferred_action_manager.h

```diff
@@ -69,6 +73,20 @@ class DeferredActionManager {
     return processed;
   }
 
+  /**
+   * Invokes GC and log manager enough times to fully GC any outstanding transactions and process deferred events.
+   * Currently, this must be done 3 times. The log manager must be called because transactions can only be GC'd once
+   * their logs are persisted.
+   * @param gc gc to use for garbage collection
+   * @param log_manager log manager to use for flushing logs
+   */
+  void FullyPerformGC(storage::GarbageCollector *const gc, storage::LogManager *const log_manager) {
```

Comment:
Is this the right place for this @GustavoAngulo? What about StorageUtil?

Reply:
I talked to @yash620 about this. His take is that FullyPerformGC should be something that is bound to an instance of the Garbage Collector. So it should be a member variable of either the GC or the DeferredActionManager.